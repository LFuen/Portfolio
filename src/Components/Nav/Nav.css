
.navigation {
    display: flex;
    justify-content: center; 
}

.navigation a {
    color:rgb(206, 206, 206);
    font-size: 40px;
    text-decoration: none;
    text-shadow: 2px 2px 5px #90e0ef;
    font-family: 'Big Shoulders Stencil Text', cursive;
    padding: 15px;
}

.link:visited {
    color:rgb(206, 206, 206);
    text-shadow: #1dcaff;
}

.link:active {
    color:rgb(206, 206, 206);
    text-shadow: #7a1212;
}

.link:hover, .link:focus {
    animation-name: buzz;
    animation-duration: 0.75s;
    animation-timing-function: linear;
    animation-iteration-count: 1;
    color: #1dcaff;
    text-shadow: 2px 2px 6px rgb(207, 111, 0);
}


@keyframes buzz {
    10% {
        -webkit-transform: translateX(3px) rotate(2deg);
        transform: translateX(3px) rotate(2deg);
    }
    
    20% {
        -webkit-transform: translateX(-3px) rotate(-2deg);
        transform: translateX(-3px) rotate(-2deg);
    }
    30% {
        -webkit-transform: translateX(3px) rotate(2deg);
        transform: translateX(3px) rotate(2deg);
    }
    40% {
        -webkit-transform: translateX(-3px) rotate(-2deg);
        transform: translateX(-3px) rotate(-2deg);
    }
    50% {
        -webkit-transform: translateX(2px) rotate(1deg);
        transform: translateX(2px) rotate(1deg);
    }
    60% {
        -webkit-transform: translateX(-2px) rotate(-1deg);
        transform: translateX(-2px) rotate(-1deg);
    }
    70% {
        -webkit-transform: translateX(2px) rotate(1deg);
        transform: translateX(2px) rotate(1deg);
    }
    80% {
        -webkit-transform: translateX(-2px) rotate(-1deg);
        transform: translateX(-2px) rotate(-1deg);
    }
    90% {
        -webkit-transform: translateX(1px) rotate(0);
        transform: translateX(1px) rotate(0);
    }
    100% {
        -webkit-transform: translateX(-1px) rotate(0);
        transform: translateX(-1px) rotate(0);
    }
}