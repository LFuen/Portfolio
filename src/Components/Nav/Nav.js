import React, {Component} from 'react'
import { Route, BrowserRouter, NavLink } from 'react-router-dom'
import './Nav.css'
import Welcome from '../Welcome/Welcome'
import About from '../About/About'
import Projects from '../Projects/Projects'
import Contact from '../Contact/Contact'
import Inquiries from '../Inquiries/Inquiries'


class Nav extends Component{
    render(){
        return(
            <BrowserRouter>
                <nav className='navigation topnav'>
                    <NavLink className='link' to='/Portfolio'>Welcome</NavLink>
                    <NavLink className='link' to='/About'>About</NavLink>
                    <NavLink className='link' to='/Projects'>Projects</NavLink>
                    <NavLink className='link' to='/Contact'>Contact</NavLink>
                    <NavLink className='link' to='/Inquiries'>Inquiries</NavLink>
                </nav>
                <Route exact path='/Portfolio' component={Welcome}/>
                <Route exact path='/About' component={About}/>
                <Route exact path='/Projects' component={Projects}/>
                <Route exact path='/Contact' component={Contact}/>
                <Route exact path='/Inquiries' component={Inquiries}/>
            </BrowserRouter>
        )
    }
}


export default Nav