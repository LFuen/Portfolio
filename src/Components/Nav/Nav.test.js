import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Nav from './Nav'


describe(`Snapshot`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const nav = renderer.create(
                <BrowserRouter>
                    <Nav />
                </BrowserRouter>).toJSON()
            expect(nav).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div')

            ReactDOM.render(
                <BrowserRouter>
                    <Nav />
                </BrowserRouter>,
                div
            )
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})