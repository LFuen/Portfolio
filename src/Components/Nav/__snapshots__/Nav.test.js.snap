// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Snapshot renders the UI as expected 1`] = `
<nav
  className="navigation"
>
  <a
    aria-current={null}
    className="link"
    href="/Portfolio"
    onClick={[Function]}
  >
    Welcome
  </a>
  <a
    aria-current={null}
    className="link"
    href="/About"
    onClick={[Function]}
  >
    About
  </a>
  <a
    aria-current={null}
    className="link"
    href="/Projects"
    onClick={[Function]}
  >
    Projects
  </a>
  <a
    aria-current={null}
    className="link"
    href="/Contact"
    onClick={[Function]}
  >
    Contact
  </a>
  <a
    aria-current={null}
    className="link"
    href="/Inquiries"
    onClick={[Function]}
  >
    Inquiries
  </a>
</nav>
`;
